internal/controller/atlasproject/atlasproject_controller.go

```diff
@@ -348,18 +347,24 @@ func logIfWarning(ctx *workflow.Context, result workflow.Result) {
 	}
 }
 
-func lastSpecFrom(atlasProject *akov2.AtlasProject, annotation string) (*akov2.AtlasProjectSpec, error) {
-	var lastApplied akov2.AtlasProject
-	ann, ok := atlasProject.GetAnnotations()[annotation]
-
-	if !ok {
-		return nil, nil
-	}
-
-	err := json.Unmarshal([]byte(ann), &lastApplied.Spec)
+func (r *AtlasProjectReconciler) clearLastAppliedMigratedResources(ctx context.Context, atlasProject *akov2.AtlasProject) error {
+	lastCfg, err := customresource.ParseLastConfigApplied(&akov2.AtlasProjectSpec{}, atlasProject)
 	if err != nil {
-		return nil, fmt.Errorf("error reading AtlasProject Spec from annotation [%s]: %w", annotation, err)
+		return fmt.Errorf("failed to parse last applied config annotation: %w", err)
 	}
-
-	return &lastApplied.Spec, nil
+	// clear all resources migrated as independent CRDs to avoid eager
+	// reconciliation that might conflict with independent CRs and apply
+	lastCfg.CustomRoles = nil
+	lastCfg.PrivateEndpoints = nil
+	lastCfg.ProjectIPAccessList = nil
+	lastCfg.NetworkPeers = nil
+	shallowCopy := akov2.AtlasProject{
+		TypeMeta:   atlasProject.TypeMeta,
+		ObjectMeta: atlasProject.ObjectMeta,
+		Spec:       *lastCfg,
+	}
+	if err := customresource.ApplyLastConfigApplied(ctx, &shallowCopy, r.Client); err != nil {
```

Comment:
You started using the full resource instead of only the spec as it id currently. This will break existing resources.

Reply:
Note this `ApplyLastConfigApplied` was always receiving the full `customresource`, not just the `spec`. Then inside the method only the spec is saved:

```go
func ApplyLastConfigApplied(ctx context.Context, resource api.AtlasCustomResource, k8sClient client.Client) error {
	uObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(resource)
	if err != nil {
		return err
	}

	js, err := json.Marshal(uObj["spec"])
	if err != nil {
		return err
	}
...
	annotations[AnnotationLastAppliedConfiguration] = string(js)
	resourceCopy.SetAnnotations(annotations)
	err = k8sClient.Patch(ctx, resourceCopy, client.MergeFrom(resource))
	if err != nil {
		return err
	}

	// retains current behavior
	resource.SetAnnotations(resourceCopy.GetAnnotations())
	return nil
}
```

Reply:
Pointed out by @s-urbaniak I will be revamping this invocation, we cannot use `ApplyLastConfigApplied` for this as it touches more things than just the annotation. For the new method I will pass the spec alone most likely.